### ISSUE TITLE
clean parameter do not behave correctly in case of multiple options

### ISSUE THREAD

**Issue by yekver** [anymaniax/orval#333]
If the `clean` parameter was provided (via cli `--clean` or as a `config.output.clean` option) the output will be unexpectedly be cleaned after each provided config option. But in `index.ts` file all the generated filed will be listed.

It is a bit tricky thing because if I use cli option `--clean` it is expected that output directory will be cleaned once but if the `clean option is provided in `output` section of each/some config options then the behaviour may be interpreted by user in a wrong way.

For example:
```
defineConfig({
 option1: { output: { clear: true } },
 option2: { output: { clear: false } },
 option3: { output: { clear: true } },
 })
 ```

**Comment by anymaniax** [comment]
Totally agree. I need to work to improve that when I have a bit of time. To give you a bit of context, it's really a first version of the feature, and didn't have a lot of time to work on it but would be great to only clean the created files. The problem is that I don't what was created in the previous run.

**Comment by yekver** [comment]
Looks like it's very hard or even impossible to track all the created files and remove them. I'd recommend easier way: store all the created files in a separate (user defined) folder and and clear it. Like a webpack + dist folder. Also definition of `clear` as an `output` parameter makes no sense right now and should be defined on configs root level.

**Comment by anymaniax** [comment]
yes, also the tsconfig option need to be moved at the root. And that what I thought for the solution but need to do it 😛
